### PR TITLE
JST-683: Golem inspect

### PR DESCRIPTION
**Pull request by SewerynKras**
Depends on https://github.com/golemfactory/golem-js/pull/789

If you want to try it out before the other PR is merged, checkout the branch `feature/JST-683/golem-sdk-inspect` in `golem-js` and `npm link` it to this repo

### CODE REVIEW

File: src/inspect/inspect.action.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+export async function inspectAction(type: string, id: string, options: InspectOptions) {
+  const yagna = new Yagna({
+    apiKey: options.yagnaAppkey,
+    basePath: options.yagnaUrl,
+  });
+  try {
+    await yagna.connect();
+  } catch (err) {
+    console.log(chalk.red(`Cannot connect to Yagna, check the app-key and URL`));
```
Comment (mgordel) [inline src/inspect/inspect.action.ts:55]: maybe a reason could be added here. Now `yagna.connect` throws `GolemPlatformError` so maybe it's worth adding a message at the end with the original reason...?

File: src/inspect/inspect.action.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  return {
+    currentState: state[0],
+    nextState: state[1],
+    usage,
+    agreement: await getAgreementDetails(yagna, agreementId),
+  };
+}
+
+function printSelectedColumns(obj: object, columns: string[]) {
+  if (columns[0] === "*") {
```
Comment (mgordel) [inline src/inspect/inspect.action.ts:39]: For me (macos) I have to use `--columns "*"`. Using `--column *` returns `{}`. Maybe it's worth to validate it.. ?

**Review by mgordel — APPROVED** [review]
I tested locally and it works fine

**Comment by mgordel** [comment]
Additionally, you also need to change `run-on-golem/shell-context.ts`, `await this.executor.end();` -> `await this.executor.terminate();` This was removed in the new beta release.

File: src/inspect/inspect.action.ts (hunk 2 above)
Comment (SewerynKras) [reply src/inspect/inspect.action.ts:39]: This is a feature of the shell you're using, if you don't wrap * in quotation marks it gets replaced with all files in the current dir (try `echo *`), there is nothing we can do about that from the cli perspective :/

**Comment by SewerynKras** [comment]
@mgordel thanks for the review, I addressed your comments, can you also take a look at the other part of this PR in golem-js: https://github.com/golemfactory/golem-js/pull/789 ?

**Review by mgordel — APPROVED** [review]

**Pull request merged**